debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotContext.java

```diff
@@ -78,6 +80,10 @@
 
     private boolean schemaVerificationPassed;
 
+    private boolean stopRequested;
```

Comment:
When is this flag cleared?

Reply:
Great catch; I think adding a reset back to `false` inside `addDataCollectionNamesToSnapshot` is sufficient.  Would you agree?

Reply:
I reworked this entirely.